packages/xrpl/test/integration/transactions/ammClawback.test.ts

```diff
@@ -0,0 +1,50 @@
+import { AMMClawback, AMMDeposit, AMMDepositFlags } from 'xrpl'
+
+import serverUrl from '../serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { createAMMPool, testTransaction } from '../utils'
+
+describe('AMMClawback', function () {
+  let testContext: XrplIntegrationTestContext
+
+  beforeAll(async () => {
+    testContext = await setupClient(serverUrl)
+  })
+  afterAll(async () => teardownClient(testContext))
+
+  it('base', async function () {
+    const ammPool = await createAMMPool(testContext.client, true)
+    const { asset, asset2, issuerWallet } = ammPool
+    const holderWallet = ammPool.lpWallet
+
+    const ammDepositTx: AMMDeposit = {
+      TransactionType: 'AMMDeposit',
+      Account: holderWallet.classicAddress,
+      Asset: asset,
+      Asset2: asset2,
+      Amount: {
+        currency: 'USD',
+        issuer: issuerWallet.address,
+        value: '10',
+      },
+      Flags: AMMDepositFlags.tfSingleAsset,
+    }
+
+    await testTransaction(testContext.client, ammDepositTx, holderWallet)
+
+    const ammClawback: AMMClawback = {
+      TransactionType: 'AMMClawback',
+      Account: issuerWallet.address,
+      Holder: holderWallet.address,
+      // @ts-expect-error -- this is an IssuedCurrency
+      Asset: asset,
```

Comment:
I don't understand this test. How does it not conflict with this validation check?

```
if (tx.Amount.issuer !== tx.Asset.issuer) {
      throw new ValidationError(
        'AMMClawback: issuer must be identical for both Amount and Asset',
      )
```

`tx.Asset` is `{'currency': 'XRP'}`. It does not have an `issuer` specified right?

Does this transaction succeed if you specify the `Amount` field?

Reply:
You're right, I just fixed it.